### PR TITLE
Fixed integer mismatch

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Fix mismatch of integral types. Followup of https://github.com/arangodb/arangodb/pull/10731.

- [X] Bug-Fix for *devel-branch*
- [ ] Bug-Fix for *3.6* (**TODO**)

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/Windows/job/arangodb-matrix-pr-windows-onlycompile/2288/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7772/

File: arangod/Agency/Node.h [hunk 1]
```diff
@@ -336,6 +336,18 @@ class Node {
   /// @brief Get double value (throws if type NODE or if conversion fails)
   double getDouble() const;
 
+  template<typename T>
```
Comment (maierlars) [inline arangod/Agency/Node.h]: :+1:

File: arangod/Agency/Node.cpp [hunk 2]
```diff
@@ -1148,3 +1138,14 @@ void Node::clear() {
     return _parent->removeChild(_nodeName);
   }
 }
+
+auto Node::getUIntAsIntWithDefault(Slice slice, std::string_view key, std::int64_t def)
+    -> std::int64_t {
+  if (slice.isObject()) {
+    Slice value = slice.get(key);
+    if (value.isUInt()) {
```
Comment (jsteemann) [inline arangod/Agency/Node.cpp]:
```suggestion
    if (value.isInt()) {
```

File: arangod/Agency/Node.h [hunk 3]
```diff
@@ -336,6 +336,20 @@ class Node {
...
+  auto getNumberUnlessExpiredWithDefault() -> T {
+    if (ADB_UNLIKELY(!lifetimeExpired())) {
```
Comment (jsteemann) [inline arangod/Agency/Node.h]:
```suggestion
    if (ADB_LIKELY(!lifetimeExpired())) {
```

File: arangod/Agency/Node.cpp (hunk 2 above)
Comment (goedderz) [reply arangod/Agency/Node.cpp]: I have the feeling that this is intended, in the sense that the step may not be a negative number. I'll check that with Kaveh.

Comment (goedderz) [reply arangod/Agency/Node.cpp]: It's not, I fixed it.

File: arangod/Agency/Node.h (hunk 3 above)
Comment (ObiWahn) [reply arangod/Agency/Node.h]: https://en.cppreference.com/w/cpp/language/attributes/likely

Comment (jsteemann) [reply arangod/Agency/Node.h]: Unfortunately that is part of the standard from C++20 on only.

Comment (ObiWahn) [reply arangod/Agency/Node.h]: Oh I forgot that it is not available here:)

**Comment by goedderz** [comment]
Jenkins is blue: http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7779/

**Comment by goedderz** [comment]
Jenkins is blue: http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7779/

**Pull request merged**